### PR TITLE
unitTesting for unzipping and enabling

### PR DESCRIPTION
**Pull request by vrdmr**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,6 +6,7 @@ BUNDLE=guest-configuration-extension.zip
 bundle: clean binary
 	@mkdir -p $(BUNDLEDIR)
 	zip ./$(BUNDLEDIR)/$(BUNDLE) ./$(BINDIR)/$(BIN)
+	zip ./$(BUNDLEDIR)/$(BUNDLE) ./agent/GCAgentx64.zip
```
Comment (vrdmr) [inline Makefile]: This is dummy for now and will be replaced once we get the final zip.

File: main/cmds_test.go [hunk 2]
```diff
@@ -1,3 +1,149 @@
...
+	require.False(t, cmds["install"].shouldReportStatus, "install should not report status")
+	require.False(t, cmds["uninstall"].shouldReportStatus, "uninstall should not report status")
+
+	// these subcommands SHOULD report status
+	require.True(t, cmds["enable"].shouldReportStatus, "enable should report status")
+	require.True(t, cmds["disable"].shouldReportStatus, "disable should report status")
+	require.True(t, cmds["update"].shouldReportStatus, "update should report status")
+}
+
+func Test_checkAndSaveSeqNum_fails(t *testing.T) {
```
Comment (vrdmr) [inline main/cmds_test.go]: @mehaag is there a pass case as well for this? Checking if the SeqNum file exists. You can try looking at Mocking the file and test for a success case.

File: main/cmds_test.go [hunk 3]
```diff
@@ -1,3 +1,149 @@
...
+}
+
+func Test_unzip_pass(t *testing.T) {
+	filenames, err := unzip(log.NewNopLogger(), "../agent/GCAgentx64.zip", "agent")
+	require.Nil(t, err)
+	require.NotEmpty(t, filenames)
+	t.Log(filenames)
+}
+
+/*
```
Comment (vrdmr) [inline main/cmds_test.go]: Delete this comment.

File: main/main.go [hunk 4]
```diff
@@ -35,6 +35,9 @@ var (
 	// downloadDir is where we store the downloaded files in the "{downloadDir}/{seqnum}/file"
 	// format and the logs as "{downloadDir}/{seqnum}/std(out|err)". Stored under dataDir
 	downloadDir = "download"
+
+
+	agentZip = "../GCAgentx64.zip"
```
Comment (vrdmr) [inline main/main.go]: It would be in a directory. Correct the path.

Comment (mehaag) [reply main/main.go]: done

File: main/cmds_test.go (hunk 2 above)
Comment (mehaag) [reply main/cmds_test.go]: The pass case is below the fail case

File: main/cmds_test.go [hunk 5]
```diff
@@ -115,6 +116,27 @@ func Test_runCmd_pass(t *testing.T) {
 	require.Nil(t, err)
 }
 
+func Test_runCmd_withTestFile(t *testing.T) {
+	dir := filepath.Join(dataDir, "testing")
+	_, err := unzip(log.NewNopLogger(), "../testing/testing.zip", dataDir)
+
+	// print files in directory
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
```
Comment (boumenot) [inline main/cmds_test.go]: If you couldn't read the directory doesn't that imply an issue?  Does it make sense to proceed?

Comment (mehaag) [reply main/cmds_test.go]: fixed

File: main/cmds_test.go [hunk 6]
```diff
@@ -122,9 +122,8 @@ func Test_runCmd_withTestFile(t *testing.T) {
 
 	// print files in directory
 	files, err := ioutil.ReadDir(dir)
-	if err != nil {
-		t.Log(err)
-	}
+	t.Log(err)
```
Comment (boumenot) [inline main/cmds_test.go]: I think the better pattern is

```golang
if err != nil {
    t.Fail(err)
}
```

You only log in the case of an error, and if there is an error it is fatal to the test.

Comment (mehaag) [reply main/cmds_test.go]: oh, I see. thank you!

Comment (mehaag) [reply main/cmds_test.go]: Fail does not take any parameters

Comment (mehaag) [reply main/cmds_test.go]: nvm, fatal works

**Pull request merged**